UnitTests/UnitTests.HighPerformance.Shared/Extensions/Test_IMemoryOwnerExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.Toolkit.HighPerformance.Buffers;
+using Microsoft.Toolkit.HighPerformance.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace UnitTests.HighPerformance.Extensions
+{
+    [TestClass]
+    public class Test_IMemoryOwnerExtensions
+    {
+        [TestCategory("IMemoryOwnerExtensions")]
+        [TestMethod]
+        public void Test_IMemoryOwnerExtensions_EmptyIMemoryOwnerStream()
+        {
+            MemoryOwner<byte> buffer = MemoryOwner<byte>.Empty;
+
+            Stream stream = buffer.AsStream();
+
+            Assert.IsNotNull(stream);
+            Assert.AreEqual(stream.Length, buffer.Length);
+            Assert.IsTrue(stream.CanWrite);
```

Comment:
We should probably validate that the stream is indeed empty without any data, right?

Likewise, we should also have tests where the original buffer contains some data, and we validate after conversion to a stream that the same data can be read back?

(We can do this in follow-on PR)

Reply:
Absolutely! Done both in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3273/commits/e8a9534595e0b019fa87fd316e8fe15171bc1f9a.